scripts/generate_rust_analyzer.sh

```diff
@@ -0,0 +1,94 @@
+#!/usr/bin/env bash
+
+srctree=$1
+objtree=$2
+lib_src=$3
+bindgen_file=$4
+
+dylib_and_extra=',"proc_macro_dylib_path":"./rust/libmodule.so","env":{"RUST_BINDINGS_FILE":"'$bindgen_file'"},
```

Comment:
Shouldn't `proc_macro_dylib_path` be only set for the `module` crate? From the [docs](https://github.com/rust-analyzer/rust-analyzer/blob/master/docs/user/manual.adoc#non-cargo-based-projects):
```typescript
    /// For proc-macro crates, path to compiled
    /// proc-macro (.so file).
    proc_macro_dylib_path?: string;
```

Reply:
Im not completely sure. I don't think so, as kernel reexports those macros, for example every driver uses module, which is in that dylib

Reply:
Hm, at least on my machine a config like this also seems to properly evaluate the proc-macro (e.g. `__nvme_rust_author` can be found in the autocompletion and the [incorrect-ident-case](https://rust-analyzer.github.io/manual.html#incorrect-ident-case) diagnostic fires inside the generated `module!` code):
```json
{
    "sysroot_src": "/home/bobo1239/.rustup/toolchains/nightly-2021-02-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library",
    "crates": [
        {
            "root_module": "rust/module.rs",
            "edition": "2018",
            "deps": [],
            "proc_macro_dylib_path": "rust/libmodule.so"
        },
        {
            "root_module": "rust/kernel/lib.rs",
            "edition": "2018",
            "deps": [
                {
                    "crate": 0,
                    "name": "module"
                }
            ]
        },
        {
            "root_module": "drivers/nvme_rust/nvme.rs",
            "edition": "2018",
            "deps": [
                {
                    "crate": 0,
                    "name": "module"
                },
                {
                    "crate": 1,
                    "name": "kernel"
                }
            ],
            "env": {"RUST_MODFILE": "???"}
        }
    ]
}
```

Reply:
Ok, I will remove the dylib for the normal modules again. I would still like to keep the custom alloc and core, to make changing them later easier.

Reply:
I'm adding the env variables to every crate (even core and alloc) I don't see a harm in that, and it makes it easier.